python_ta/debug/accumulation_table.py

```diff
@@ -0,0 +1,158 @@
+    loop_var_name: str
+    loop_var_val: list
+    _loop_lineno: int
+    _zero_iteration: bool
+
+    def __init__(self, accumulation_names: list) -> None:
+        self.loop_accumulators = {accumulator: [] for accumulator in accumulation_names}
+        self.loop_var_name = ""
+        self.loop_var_val = []
+        self._loop_lineno = 0
+        self._zero_iteration = True
+
+    def get_lineno(self, line_no: int) -> None:
+        """Get the line number of the for loop nested in the with statement"""
+        self._loop_lineno = line_no
+
+    def _add_iteration(self, val: list, var: Any) -> None:
+        """Add the values of the accumulator and loop variables of an iteration"""
+        self.loop_var_val.append(var)
+        for index, key in enumerate(self.loop_accumulators):
+            self.loop_accumulators[key].append(val[index])
+
+    def _create_iteration_dict(self) -> dict:
+        """Use the values of each iteration and return a dictionary that maps each accumulator
+        and loop variable to its respective value during each iteration
+        """
+        return {
+            "iteration": list(range(len(self.loop_accumulators[list(self.loop_accumulators)[0]]))),
+            "loop variable (" + self.loop_var_name + ")": self.loop_var_val,
+            **self.loop_accumulators,
+        }
+
+    def _tabulate_data(self) -> None:
+        """Print the values of the accumulator and loop variables into a table"""
+        iteration_dict = self._create_iteration_dict()
+        print(
+            tabulate.tabulate(
+                iteration_dict, headers="keys", colalign=(*["left"] * len(iteration_dict),)
+            )
+        )
+
+    def add_zero_iter(self, func_node: types.NodeNG, func_frame: types.FrameType) -> None:
+        """Add the names and values of the all the accumulators for the zeroth iteration"""
+        self.loop_var_name = func_node.target.name
+        for name in self.loop_accumulators:
+            if name in func_frame.f_locals:
+                self.loop_accumulators[name] = [func_frame.f_locals[name]]
+            else:
+                raise NameError
+
+    def trace_loop(self, frame: types.FrameType, event: str, arg: Any) -> None:
+        """Trace through the for loop and store the values of the
+        accumulators during each iteration
+        """
+        if event == "line" and frame.f_lineno == self._loop_lineno:
+            local_vars = frame.f_locals
+            curr_vals = []
+            for accumulator in self.loop_accumulators:
+                if accumulator in local_vars:
+                    curr_vals.append(local_vars[accumulator])
+                else:
+                    raise NameError
+
+            if self.loop_var_name in local_vars and not self._zero_iteration:
+                self._add_iteration(curr_vals, local_vars[self.loop_var_name])
+
+            self._zero_iteration = False
+
+    def __enter__(self) -> AccumulationTable:
```

Comment:
Include a docstring to specify what happens when the "with context" is entered. You may wish to combine some parts of `setup_table` with this.